### PR TITLE
ref(onboarding): update `setupMessagingIntegrationButton` API queries

### PR DESCRIPTION
**Pull request by ameliahsu**
updated `setupMessagingIntegrationButton` to use `OrganizationIntegrationsEndpoint` instead of `ProjectDetailsEndpoint` to determine if the organization has any messaging integrations already installed. we need to do this because we plan on adding this button to the Project Creation page where a project does not exist yet

also moved queries to `OrganizationConfigIntegrationsEndpoint` to the button level instead of the modal level, so that if the API requests fail then the button will fail to render, rather than the modal instantly closing upon failure which looked kinda bad 😳

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
This all looks reasonable to me, but you might want a second review from someone who is more familiar with this!

**Review by iamrajjoshi — APPROVED** [review]
looks gud, gud job

**Pull request merged**
